packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -87,6 +93,8 @@ export interface DecorateQueryProcedure<TDef extends ResolverDef> {
    */
   queryOptions: TRPCQueryOptions<TDef>;
 
+  useQuery: TRPCUseQueryResult<TDef>;
```

Comment:
We could flag this so it's not exposed by default, and people coming from the old API could enable to have a more familiar API.

Something like:

```ts
createTRPCOptionsProxy<AppRouter, {
  enableOldHooks: true
}>()
```

Reply:
Yeah I guess, but as you pointed out in https://github.com/trpc/trpc/discussions/6508#discussioncomment-12280845 there is already away to get the old API, if it is indeed considered "old". My hope with this is to figure out if we can keep around the old API for most use cases indefinitely, because it's better DX. But if that were the direction making it opt in and using the label "old" wouldn't make sense.

If we'd just be adding it to help people migrate extra slowly, then yeah that could be sensible.